File: go/cell.go
package main

type Cell struct {
  x uint32
  y uint32
  alive bool
  nextState bool
  neighbours []*Cell
}

func newCell(x uint32, y uint32, alive bool) *Cell {
  return &Cell{x: x, y: y, alive: alive}
}

func (cell *Cell) toChar() rune {
  if cell.alive {
    return 'o'
  } else {
    return ' '
  }
}

func (cell *Cell) aliveNeighbours() uint32 {
  // The following is the fastest
  aliveNeighbours := uint32(0)
  for _, neighbour := range cell.neighbours {
    if neighbour.alive {
      aliveNeighbours++
    }
  }
  return aliveNeighbours

  // The following is slower
  // aliveNeighbours := uint32(0)
  // count := len(cell.neighbours)
  // for i := range count {
  //   neighbour := cell.neighbours[i]
  //   if neighbour.alive {
  //     aliveNeighbours++
  //   }
  // }
  // return aliveNeighbours
}

File: python/cell.py
from dataclasses import dataclass, field

@dataclass
class Cell:
  x: int
  y: int
  alive: bool = False
  next_state: bool | None = None
  neighbours: list["Cell"] = field(default_factory=list)

  def to_char(self) -> str:
    return 'o' if self.alive else ' '

  def alive_neighbours(self) -> int:
    # The following is slower
    # filter_alive = lambda neighbour: neighbour.alive
    # return len(list(filter(filter_alive, self.neighbours)))

    # The following is the fastest
    alive_neighbours = 0
    for neighbour in self.neighbours:
      if neighbour.alive:
        alive_neighbours += 1
    return alive_neighbours

    # The following is slower
    # alive_neighbours = 0
    # count = len(self.neighbours)
    # for i in range(count):
    #   neighbour = self.neighbours[i]
    #   if neighbour.alive:
    #     alive_neighbours += 1
    # return alive_neighbours
